src/chaos/controller.go

```diff
@@ -0,0 +1,210 @@
+	if err != nil {
+		return nil, err
+	}
+
+	alivePods := make([]string, 0)
+	for _, pod := range podList.Items {
+		alivePods = append(alivePods, pod.GetName())
+	}
+
+	return alivePods, nil
+}
+
+func (cc *ChaosController) deleteNode(namespace string, podName string) error {
+	podGVR := schema.GroupVersionResource{
+		Group:    "", // Core API group
+		Version:  "v1",
+		Resource: "pods",
+	}
+
+	return cc.DynamicClient.Resource(podGVR).Namespace(namespace).Delete(context.TODO(), podName, v1.DeleteOptions{})
+}
+
+// configConfigPath    - path to k8s cluster config
+// killNodes 	- to kill or not to kill affected nodes after each run
+func RunChaosExperiment(namespace string, chaosConfigPath string) error {
+	cc := NewController(namespace)
+
+	// Work with experiment yaml file
+	experimentFile, err := os.ReadFile(chaosConfigPath)
+	if err != nil {
+		log.Fatalf("Failed to read manifest file: %v", err)
+		return err
+	}
+	reader := bytes.NewReader(experimentFile)
+	decoder := yaml.NewYAMLOrJSONDecoder(reader, 1000)
+	parsedYaml := &unstructured.Unstructured{}
+
+	if err := decoder.Decode(parsedYaml); err != nil {
+		log.Fatalf("Failed to decode YAML: %v", err)
+		return err
+	}
+
+	expType, _, err := unstructured.NestedString(parsedYaml.Object, "kind")
+	if err != nil {
+		return err
+	}
+	gvr := schema.GroupVersionResource{
+		Group:    "chaos-mesh.org",         // ChaosMesh group
+		Version:  "v1alpha1",               // Version of the ChaosMesh resource
+		Resource: strings.ToLower(expType), // Resource is "podchaos"
+	}
+
+	// Apply the resource (equivalent to `kubectl apply -f`)
+	_, err = cc.DynamicClient.Resource(gvr).Namespace(cc.Namespace).Create(context.TODO(), parsedYaml, v1.CreateOptions{})
+	if err != nil {
+		log.Fatalf("Failed to apply ChaosMesh resource: %v", err)
+	}
+
+	return err
+}
+
+// Deleting the experiment from chaosmesh list
+// to avoid collisions, when the same experiment is started
+func ClearChaosCache(namespace string) error {
+	cc := NewController(namespace)
+	var chaosTypes = []string{"podchaos", "networkchaos"}
+
+	for _, chaosType := range chaosTypes {
+		gvr := schema.GroupVersionResource{
+			Group:    "chaos-mesh.org", // ChaosMesh group
+			Version:  "v1alpha1",       // Version of the ChaosMesh resource
+			Resource: chaosType,
+		}
+
+		err := cc.DynamicClient.Resource(gvr).Namespace(cc.Namespace).DeleteCollection(
+			context.TODO(),
+			v1.DeleteOptions{},
+			v1.ListOptions{},
+		)
+		if err != nil {
+			log.Fatalf("Failed to delete Network resources: %v", err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Check the health of each node in cluster
+// Returning string so that JS can handle it correctly
+func CheckClusterHealth(namespace string) error {
+	cc := NewController(namespace)
+
+	allPods, err := cc.listNodes(namespace)
+	if err != nil {
+		return err
+	}
+
+	for _, podName := range allPods {
+		if err := cc.checkSingleNodeHealth(podName); err != nil {
+			return err
+		}
+	}
+
+	return nil
```

Comment:
Maybe it's more logical to return `true, nil` here?

Reply:
The result, visible to user will be a structure {true, nil}. Return statement, visible for users are define in file khorne.go
```func (p *Chaos) CheckClusterHealth(namespace string) Result {
	if err := chaos.CheckClusterHealth(namespace); err != nil {
		return Result{false, err.Error()}
	}
	return Result{Success: true}
}
```